aodndata/acorn/handler.py

```diff
@@ -0,0 +1,69 @@
+import os
+import re
+from datetime import datetime
+
+from aodncore.pipeline import HandlerBase
+from aodncore.pipeline.exceptions import InvalidFileNameError
+
+
+FILE_TYPE_NEED_INDEX = ['radial', 'radial_quality_controlled', 'gridded_1h-avg-current-map_non-QC',
+                                'gridded_1h-avg-current-map_QC', 'gridded_1h-avg-wind-map_QC',
+                                'gridded_1h-avg-wave-site-map_QC']
+
+FILE_TYPES = '|'.join(FILE_TYPE_NEED_INDEX)
+
+
+def get_type(filepath):
+    """return acorn_file_type, the file type of an ACORN file based on its filename"""
+
+    if "FV00_radial.nc" in filepath:
+        acorn_file_type = "radial"
+    elif "FV00_sea-state.nc" in filepath:
+        acorn_file_type = "vector"
+    elif "FV01_radial.nc" in filepath:
+        acorn_file_type = "radial_quality_controlled"
+    elif "FV01_wavespec.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-wave-spectra_QC"
+    elif "FV01_windp.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-wind-map_QC"
+    elif "FV01_wavep.nc" in filepath:
+        site_map_station = ('_CBG_', '_SAG_', '_ROT_', '_COF_')
+        if any(s in filepath for s in site_map_station):
+            acorn_file_type = "gridded_1h-avg-wave-site-map_QC"
+        else:
+            acorn_file_type = "gridded_1h-avg-wave-station-map_QC"
+    elif "FV00_1-hour-avg.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-current-map_non-QC"
+    elif "FV01_1-hour-avg.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-current-map_QC"
+    else:
+        raise InvalidFileNameError("file name: \"{filename}\" not matching regex to deduce dest_path".
+                                   format(filename=os.path.basename(filepath)))
+
+    return acorn_file_type
+
+
+class AcornHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(AcornHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = ['.nc']
+
+    @staticmethod
+    def dest_path(filepath):
+        m = re.search(r'^IMOS_ACORN_([A-Z].*)_([0-9]{8}T[0-9]{6}Z)_([A-Z].*)_(FV0[0-2])_(.*)\.nc$', filepath)
```

Comment:
Original bash regex is as follow:
```Bash
ACORN_RT_REGEX='^IMOS_ACORN_[[:alpha:]]{1,2}_[[:digit:]]{8}T[[:digit:]]{6}Z_[[:alpha:]]{3,4}_FV00_(radial|sea-state|wavespec|windp|wavep|1-hour-avg)\.nc$'
ACORN_DM_REGEX='^IMOS_ACORN_[[:alpha:]]{1,2}_[[:digit:]]{8}T[[:digit:]]{6}Z_[[:alpha:]]{3,4}_FV01_(radial|sea-state|wavespec|windp|wavep|1-hour-avg)\.nc$'
```

Can we keep the constraint on the site / station code limited to 3 or 4 characters? Also there is no FV02 allowed, only FV00 and FV01.